cpp/src/neighbors/ivf_common.cuh

```diff
@@ -254,7 +254,14 @@ void postprocess_distances(ScoreOutT* out,      // [n_queries, topk]
         raft::linalg::unaryOp(out, in, len, raft::sqrt_op{}, stream);
       }
     } break;
-    case distance::DistanceType::CosineExpanded:
+    case distance::DistanceType::CosineExpanded: {
+      raft::linalg::unaryOp(
+        out,
+        in,
+        len,
+        raft::compose_op(raft::add_const_op<ScoreOutT>{1.0}, raft::cast_op<ScoreOutT>{}),
+        stream);
+    } break;
```

Comment:
I don't get it, why do we need `f(x) = x + 1` mapping here? I'd expect it to be the same as for inner product, or with extra diving by the query norm.

Reply:
x  is equal to the negative of the normalized inner product. We add one in the end to get the cosine distance as
`1 - (a.b) / ||a||||b||`